example/use_cases/employer_university_flow/pkg/util.go

```diff
@@ -0,0 +1,240 @@
+	e := Entity{
+		wallet: example.NewSimpleWallet(),
+		Name:   name,
+	}
+	if err := e.wallet.Init(keyType); err != nil {
+		return nil, err
+	}
+	return &e, nil
+}
+
+func GenerateDIDPeer() (did.DID, error) {
+	kt := crypto.Ed25519
+	pubKey, _, err := crypto.GenerateKeyByKeyType(kt)
+	if err != nil {
+		return nil, err
+	}
+	did, err := did.PeerMethod0{}.Generate(kt, pubKey)
+	if err != nil {
+		return nil, err
+	}
+	return did, nil
+}
+
+// This validates the VC.
+// TODO: Expand on this more
+// Simplify it?
+func validateVC(vc credential.VerifiableCredential) error {
+
+	issuer := "https://example.edu/issuers/565049"
+	var AssertionMethod cryptosuite.ProofPurpose = "assertionMethod"
+	var vc2 credential.VerifiableCredential
+	err := util.Copy(&vc, &vc2)
+	if err != nil {
+		return err
+	}
+	var OKP = cryptosuite.KTY("OKP")
+	var Ed25519 = cryptosuite.CRV("Ed25519")
+
+	jwk, err := cryptosuite.GenerateJSONWebKey2020(OKP, Ed25519)
+	if err != nil {
+		return err
+	}
+
+	signer, err := cryptosuite.NewJSONWebKeySigner(issuer, jwk.PrivateKeyJWK, AssertionMethod)
+	if err != nil {
+		return err
+	}
+
+	suite := cryptosuite.GetJSONWebSignature2020Suite()
+	err = suite.Sign(signer, &vc2)
+	if err != nil {
+		return err
+	}
+	verifier, err := cryptosuite.NewJSONWebKeyVerifier(issuer, jwk.PublicKeyJWK)
+	if err != nil {
+		return err
+	}
+	err = suite.Verify(verifier, &vc2)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Build a presentation request (PR)
+// A PR is sent by a holder to a verifier
+// It can be sent over multiple mechanisms
+// For more information, please go to here:
+// https://identity.foundation/presentation-exchange/#presentation-request
+// and for the source code with the sdk,
+// https://github.com/TBD54566975/ssi-sdk/blob/main/credential/exchange/request.go
+// is appropriate to start off with.
+func makePresentationRequest(jwk cryptosuite.JSONWebKey2020, presentationData exchange.PresentationDefinition, targetId string) (pr []byte, signer *cryptosuite.JSONWebKeySigner, err error) {
+
+	example.CustomWriter.WriteNote("Presentation Request (JWT) is created")
+
+	// Signer:
+	// https://github.com/TBD54566975/ssi-sdk/blob/main/cryptosuite/jsonwebkey2020.go#L350
+	// Implements: https://github.com/TBD54566975/ssi-sdk/blob/main/cryptosuite/jwt.go#L12
+	signer, err = cryptosuite.NewJSONWebKeySigner(jwk.ID, jwk.PrivateKeyJWK, cryptosuite.Authentication)
+	if err != nil {
+		return
+	}
+
+	// Builds a presentation request
+	// Requires a signeer, the presentation data, and the target
+	// Target is the Audience Key
+	requestJWTBytes, err := exchange.BuildJWTPresentationRequest(*signer, presentationData, targetId)
+	if err != nil {
+		return
+	}
+
+	return requestJWTBytes, signer, err
+}
+
+func MakePresentationRequest(jwk cryptosuite.JSONWebKey2020, presentationData exchange.PresentationDefinition, targetId string) (pr []byte, signer *cryptosuite.JSONWebKeySigner, err error) {
```

Comment:
What is the point of this?

Reply:
I'm assuming the question is why private and public mirror functions. So the idea was originally keeping everything private. Then I realized that I wanted to make some public, but I didn't want to refactor the private function calls. I've also seen this done sometimes, to keep separation of concerns. Private for functionality, then a wrapper to expose it. If you need to hide the function again, it's trivial. 

That being said: The style of private functions with an external wrapper is probably not useful for this demo, so I've adjusted things.